src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+    // Storage
+
+    /// @dev Value for what the next id will be.
+    uint private _nextId;
+
+    /// @dev Registry mapping ids to Bounty structs.
+    mapping(uint => Bounty) private _bountyRegistry;
+
+    /// @dev List of Bounty id's.
+    LinkedIdList.List _bountyList;
+
+    /// @dev Registry mapping ids to Claim structs.
+    mapping(uint => Claim) private _claimRegistry;
+
+    /// @dev List of Claim id's.
+    LinkedIdList.List _claimList;
+
+    //@dev Connects contributor addresses to claim Ids
+    mapping(address => EnumerableSet.UintSet) contributorAddressToClaimIds;
+    //--------------------------------------------------------------------------
+    // Initialization
+
+    /// @inheritdoc Module
+    function init(IProposal proposal_, Metadata memory metadata, bytes memory)
+        external
+        override(Module)
+        initializer
+    {
+        __Module_init(proposal_, metadata);
+        //init empty list of bounties and claims
+        _bountyList.init();
+        _claimList.init();
+    }
+
+    //--------------------------------------------------------------------------
+    // Getter Functions
+
+    /// @inheritdoc IBountyManager
+    function getBountyInformation(uint bountyId)
+        external
+        view
+        validBountyId(bountyId)
+        returns (Bounty memory)
+    {
+        return _bountyRegistry[bountyId];
+    }
+
+    /// @inheritdoc IBountyManager
+    function listBountyIds() external view returns (uint[] memory) {
+        return _bountyList.listIds();
+    }
+
+    /// @inheritdoc IBountyManager
+    function isExistingBountyId(uint bountyId) public view returns (bool) {
+        return _bountyList.isExistingId(bountyId);
+    }
+
+    /// @inheritdoc IBountyManager
+    function getClaimInformation(uint claimId)
+        external
+        view
+        validClaimId(claimId)
+        returns (Claim memory)
+    {
+        return _claimRegistry[claimId];
+    }
+
+    /// @inheritdoc IBountyManager
+    function listClaimIds() external view returns (uint[] memory) {
+        return _claimList.listIds();
+    }
+
+    /// @inheritdoc IBountyManager
+    function isExistingClaimId(uint claimId) public view returns (bool) {
+        return _claimList.isExistingId(claimId);
+    }
+
+    /// @inheritdoc IBountyManager
+    function listClaimIdsForContributorAddress(address contributorAddrs)
+        external
+        view
+        returns (uint[] memory)
+    {
+        return contributorAddressToClaimIds[contributorAddrs].values();
+    }
+
+    //--------------------------------------------------------------------------
+    // Mutating Functions
+
+    /// @inheritdoc IBountyManager
+    function addBounty(
+        uint minimumPayoutAmount,
+        uint maximumPayoutAmount,
+        bytes calldata details
+    )
+        external
+        onlyRole(uint8(Roles.BountyAdmin))
+        validPayoutAmounts(minimumPayoutAmount, maximumPayoutAmount)
+        returns (uint id)
```

Comment:
Follow a single convention for return data from functions.

Reply:
What convention are you referring to?